### PR TITLE
Added test for global and repo presets defined in config.js

### LINKED ISSUE #1080: extends array doesn't work if configured in config.js
e.g. config.js = `{ extends: [ 'config:base' ] }`

The values in the `config:base` preset get overridden by the default *app* values, which aren't always the same. It's more noticeable if the preset(s) attempted modify the app defaults by a lot.

See #1055  for details of the problem it causes.

### PR DESCRIPTION
**Pull request by hulkish**
Fixes #1080

### CODE REVIEW

File: lib/config/index.js [hunk 1]
```diff
@@ -24,11 +26,20 @@ async function parseConfigs(env, argv) {
   const cliConfig = cliParser.getConfig(argv);
   const envConfig = envParser.getConfig(env);
 
+  if (Array.isArray(fileConfig.repositories)) {
```
Comment (rarkins) [inline lib/config/index.js]: Can't we just resolve `fileConfig`?

Comment (hulkish) [reply lib/config/index.js]: Well no, because `repositories` is not handled for this inside `resolveConfigPresets`.

**Comment by rarkins** [comment]
Then maybe as a hack we just resolve both? We definitely want to resolve top level extends

**Comment by hulkish** [comment]
Sure, sounds reasonable.

File: lib/config/index.js (hunk 1 above)
Comment (rarkins) [reply lib/config/index.js]: Is that itself a bug then? I would expect that we do dig into `repositories` and any other array with this line: https://github.com/renovateapp/renovate/blob/101226db977d8fa85c50bcef78f18e39735f1081/lib/config/presets.js#L78

**Review by rarkins — CHANGES_REQUESTED** [review]
Handle top-level `extends`

File: lib/config/cli.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ function getConfig(argv) {
     boolean: val => val === 'true',
     list: val => val.split(',').map(el => el.trim()),
     string: val => val,
+    integer: parseInt,
```
Comment (hulkish) [inline lib/config/cli.js]: This is a fix for a bug I discovered where integer values weren't being coerced from a string.

**Comment by hulkish** [comment]
@rarkins ok, I think this is all set

**Comment by rarkins** [comment]
@hulkish I've kind of cherry picked this PRs functionality onto master and believe #1080 is now working completely.

I'm not sure I want the code that's left here. It looks like you want to combine provisioned + autodiscovered repositories? Previously our functionality was to either/or and not combined, so that might be considered a breaking change. Please let me know if I've understood correctly.

**Comment by hulkish** [comment]
@rarkins Cool, I've reworked this pr to only include the test.

**Pull request merged**